packages/ui/src/components/MenuV2/__stories__/WithModal.stories.tsx

```diff
@@ -0,0 +1,41 @@
+import type { StoryFn } from '@storybook/react'
+import { MenuV2 } from '..'
+import { Modal } from '../../Modal'
+import { DefaultDisclosure } from './Template.stories'
+
+export const WithModal: StoryFn<typeof MenuV2> = () => {
+  const NestedModal = () => (
+    <Modal disclosure={<MenuV2.Item>MenuItem with Modal</MenuV2.Item>}>
+      <div style={{ padding: 32 }}>
+        Content should be present in center of the modal
+      </div>
+      <div style={{ padding: 32 }}>
+        Content should be present in center of the modal
+      </div>
+    </Modal>
+  )
+
+  return (
+    <MenuV2 disclosure={DefaultDisclosure}>
+      <MenuV2.Item>Menu Item</MenuV2.Item>
+      <MenuV2.Item href="/?path=/docs/components-navigation-menu--modal">
```

Comment:
It's not very important but the link seems broken

`/?/?path=/docs/components-navigation-menu"`